OrbitCaptureClient/CaptureClient.cpp

```diff
@@ -200,18 +208,23 @@ bool CaptureClient::TryAbortCapture() {
   return true;
 }
 
-void CaptureClient::FinishCapture() {
+ErrorMessageOr<void> CaptureClient::FinishCapture() {
   if (reader_writer_ == nullptr) {
-    return;
+    return outcome::success();
   }
 
   grpc::Status status = reader_writer_->Finish();
-  if (!status.ok()) {
-    ERROR("Finishing gRPC call to Capture: %s", status.error_message());
-  }
   reader_writer_.reset();
   client_context_.reset();
 
-  absl::MutexLock lock(&state_mutex_);
-  state_ = State::kStopped;
+  {
+    absl::MutexLock lock(&state_mutex_);
+    state_ = State::kStopped;
+  }
+
+  if (!status.ok()) {
```

Comment:
I think it's fine to do it under the lock, if it makes code more readable.

Reply:
I think it makes the code less readable, because it carries the question "Why does `status` need to be inspected under the lock?".

Reply:
well, the answer here is - we want state_ to become kStopped once we exit this function, not earlier :) (I know there is no state change here but still). You can keep it as it is, I do not have strong feelings one way or the other.

Reply:
Indeed I don't really agree with that answer and that's why that would make it less readable. Why do we want `state_` to become `kStopped` when we exit the function? After all, there is other stuff happening after `FinishCapture` is called, so why exactly when `FinishCapture` exits? Instead, we want `state_` to become `kStopped` after `reader_writer_.reset(); client_context_.reset();`, and having the limited scope of the lock makes it clearer.

Of course this is a minor thing but that's my interpretation of why the lock should be as limited as possible, so that it's clearer where the state change actually happens.